console-subscriber/src/lib.rs

```diff
@@ -494,6 +500,17 @@ impl<S> Layer<S> for ConsoleLayer
 where
     S: Subscriber + for<'a> LookupSpan<'a>,
 {
+    fn on_register_dispatch(&self, dispatch: &Dispatch) {
+        let tracer = consequences::Tracer::from_dispatch::<S>(dispatch);
+        if let Err(_) = self.shared.tracer.set(tracer) {
+            // XXX(jswrenn): This layer has seemingly been supplied *multiple*
+            // subscribers. Is that possible? I guess you can call
+            // `Layer::with_subscriber` multiple times; what happens if you
+            // install the resulting `Layered` as a subscriber? Is `Arc`ing a
+            // `Layered<ConsoleLayer, S>` also a recipe for weirdness?
```

Comment:
I think a `Layer` implementation for `Arc<L: Layer<_>>` is the only way this could be called twice. Fortunately, [there is no such impl in the current release version](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.Layer.html#foreign-impls)